### PR TITLE
Post Features - 1

### PR DESCRIPTION
**Pull request by SukanyaSen530**
- View all posts on feed (will later change to posts of people only user follows)
- Add post
   - Add minimum 1 photo and max 3 photos
   - Emoji Support
- Edit Post 
  - Can edit description
- Delete Post

### CODE REVIEW

File: src/components/Cards/PostCard/PostCard.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import "./post-card.scss";
+import PostCardHeader from "./PostCardHeader";
+import PostCardBody from "./PostCardBody";
+import BookMarkButton from "../../Buttons/BookMarkButton";
+import LikeButton from "../../Buttons/LikeButton";
+import ShareButton from "../../Buttons/ShareButton";
+
+const PostCard = ({
+  _id,
+  description,
```
Comment (biswanath-tewari) [inline src/components/Cards/PostCard/PostCard.jsx:15]: liked how you have kept it a reusable dumb component.

File: src/redux/slices/postSlice.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      description: "",
+      postId: 0,
+    },
+  },
+  reducers: {
+    togglePostModal: (state, action) => {
+      state.isOpen = !state.isOpen;
+    },
+    updateEditPostData: (state, action) => {
+      console.log("action", action);
```
Comment (biswanath-tewari) [inline src/redux/slices/postSlice.js]: we should not place any side effects inside a reducer.

File: src/components/Buttons/BookMarkButton.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+
+import { MdBookmarkBorder, MdBookmark } from "react-icons/md";
+
+function BookMarkButton({
+  isbookMarked = true,
+  isLoading = false,
+  handleBookMark = () => {},
+}) {
+  const handleDisable = (e) => {
```
Comment (biswanath-tewari) [inline src/components/Buttons/BookMarkButton.jsx:10]: I guess you missed invoking this function, eslint comes in handy in these situations.

**Review by biswanath-tewari — APPROVED** [review]

**Review by KanekarSnehal — APPROVED** [review]
Liked the different folder for the button component
Liked the use of RTK query
Liked how to code is structured into various components

**Review by animeshgarai09 — APPROVED** [review]
Everything looks good

File: src/components/Buttons/BookMarkButton.jsx (hunk 3 above)
Comment (SukanyaSen530) [reply src/components/Buttons/BookMarkButton.jsx:10]: I have not implement this feature yet.

File: src/redux/slices/postSlice.js (hunk 2 above)
Comment (SukanyaSen530) [reply src/redux/slices/postSlice.js]: fixed - [dbef5a1](https://github.com/SukanyaSen530/Vibes-frontend/pull/8/commits/dbef5a108bdff840f973c63ac990a81eb1276e9f)
Thanks for pointing it out!

**Pull request merged**